src/Algebra/Graph/Bipartite/AdjacencyMap.hs

```diff
@@ -0,0 +1,667 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Bipartite.AdjacencyMap
+-- Copyright  : (c) Andrey Mokhov 2016-2019
+-- License    : MIT (see the file LICENSE)
+-- Maintainer : andrey.mokhov@gmail.com
+-- Stability  : experimental
+--
+-- __Alga__ is a library for algebraic construction and manipulation of graphs
+-- in Haskell. See <https://github.com/snowleopard/alga-paper this paper> for
+-- the motivation behind the library, the underlying theory, and
+-- implementation details.
+--
+-- This module defines the 'AdjacencyMap' data type for bipartite graphs and
+-- basic associated functions.
+----------------------------------------------------------------------------
+module Algebra.Graph.Bipartite.AdjacencyMap (
+    -- * Data structure
+    AdjacencyMap, leftAdjacencyMap, rightAdjacencyMap,
+
+    -- * Basic graph construction primitives
+    empty, leftVertex, rightVertex, vertex, edge, overlay, connect,
+    vertices, edges, overlays, connects, swap,
+
+    -- * Conversion functions
+    toBipartite, fromBipartite, fromGraph,
+
+    -- * Graph properties
+    isEmpty, hasEdge, hasLeftVertex, hasRightVertex, hasVertex, leftVertexCount,
+    rightVertexCount, vertexCount, edgeCount, leftVertexList, rightVertexList,
+    vertexList, edgeList, leftVertexSet, rightVertexSet, vertexSet, edgeSet,
+
+    -- * Miscellaneous
+    consistent,
+    ) where
+
+import Data.Either (lefts, rights)
+import Data.List   (sort, (\\))
+import GHC.Generics
+
+import qualified Algebra.Graph              as G
+import qualified Algebra.Graph.AdjacencyMap as AM
+
+import qualified Data.Map.Strict as Map
+import qualified Data.Set        as Set
+import qualified Data.Tuple
+
+{-| The 'Bipartite.AdjacencyMap' data type represents an __undirected__
+bipartite graph by two maps of vertices into their adjacency sets. The two type
```

Comment:
I was told by @snowleopard  that it's best to leave out the implementation details from the documentation as the implementation might change later.

Reply:
Good point!

Note that it's not only about the implementation that might change in future, but also about unnecessary information in the documentation, which takes up the time of the reader.

Reply:
Thanks for the comment.

I added this in order to show some difference between `Algebra.Graph.Bipartite.AdjacencyMap` and `Algebra.Graph.Bipartite` (if added in future). Now I see that there is no such difference in the documentation for `Algebra.Graph.AdjacencyMap`, so I removed it.